pkg/ee/resource-quota/resource-quota-syncer/syncer.go

```diff
@@ -0,0 +1,183 @@
+//go:build ee
+
+/*
+                  Kubermatic Enterprise Read-Only License
+                         Version 1.0 ("KERO-1.0”)
+                     Copyright © 2022 Kubermatic GmbH
+
+   1.	You may only view, read and display for studying purposes the source
+      code of the software licensed under this license, and, to the extent
+      explicitly provided under this license, the binary code.
+   2.	Any use of the software which exceeds the foregoing right, including,
+      without limitation, its execution, compilation, copying, modification
+      and distribution, is expressly prohibited.
+   3.	THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package resourcequotasyncer
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	apiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/controller/util/predicate"
+	kuberneteshelper "k8c.io/kubermatic/v2/pkg/kubernetes"
+	kubermaticresources "k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// This controller syncs the ResourceQuotas from the master cluster to the seed clusters.
+	ControllerName = "kkp-resource-quota-syncer"
+)
+
+type reconciler struct {
+	log          *zap.SugaredLogger
+	masterClient ctrlruntimeclient.Client
+	namespace    string
+	seedClients  map[string]ctrlruntimeclient.Client
+	recorder     record.EventRecorder
+}
+
+func Add(masterMgr manager.Manager,
+	seedManagers map[string]manager.Manager,
+	log *zap.SugaredLogger,
+) error {
+	log = log.Named(ControllerName)
+	r := &reconciler{
+		log:          log,
+		masterClient: masterMgr.GetClient(),
+		seedClients:  map[string]ctrlruntimeclient.Client{},
+		recorder:     masterMgr.GetEventRecorderFor(ControllerName),
+	}
+
+	c, err := controller.New(ControllerName, masterMgr, controller.Options{
+		Reconciler: r,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	for seedName, seedManager := range seedManagers {
+		r.seedClients[seedName] = seedManager.GetClient()
```

Comment:
what is new seed will be added in runtime?

Reply:
This controller is a part of the seed-controller-lifecycle, so if I understand correctly, every time a new seed is added or removed, all the controllers are restarted